app/fight.py

```diff
@@ -0,0 +1,22 @@
+from app.knights import Knight
+
+
+class Arena:
+    @staticmethod
+    def opposition(knights: dict) -> dict:
+        dict_knights = {}
+        for key, value in knights.items():
+            knight = Knight(value["name"], value["power"],
+                            value["hp"])
+            knight.use_weapon(value["weapon"]["power"])
+            if value["armour"] is not None:
+                knight.use_armour(value["armour"])
+            if value["potion"] is not None:
+                knight.use_potion(value["potion"]["effect"])
+            dict_knights[key] = knight
+
+        dict_knights["lancelot"].hp -= dict_knights["mordred"].power
```

Comment:
You can create method in Arena witch get names of opponents and do 18 and 19 line (to aboid dublication)